lib/segment/src/index/sparse_index/sparse_vector_index.rs

```diff
@@ -137,16 +137,12 @@ impl<TInvertedIndex: InvertedIndex> SparseVectorIndex<TInvertedIndex> {
                     }
                     indices_tracker.register_indices(vector);
                     let vector = indices_tracker.remap_vector(vector.to_owned());
-                    ram_index.upsert(id, vector);
-                    index_point_count += 1;
+                    ram_index_builder.add(id, vector);
```

Comment:
where the rubber hits the road :motorcycle: 

instead of upserting the index with binary search, we push it into a cheap builder